examples/chimp.py

```diff
@@ -1,203 +1,234 @@
         center = self.rect.center
-        self.dizzy = self.dizzy + 12
-        if self.dizzy >= 360:
-            self.dizzy = False
-            self.image = self.original
+        self.rotation += 12
+        if self.rotation >= 360:
+            self.rotation = 0
+            self.move = self._walk
+            self.image = self.image_original
         else:
             rotate = pg.transform.rotate
-            self.image = rotate(self.original, self.dizzy)
+            self.image = rotate(self.image_original, self.rotation)
+
         self.rect = self.image.get_rect(center=center)
 
-    def punched(self):
-        """this will cause the monkey to start spinning"""
-        if not self.dizzy:
-            self.dizzy = True
-            self.original = self.image
-
-
-def main():
-    """this function is called when the program starts.
-    it initializes everything it needs, then runs in
-    a loop until the function returns."""
-    # Initialize Everything
-    pg.init()
-    screen = pg.display.set_mode((1280, 480), pg.SCALED)
-    pg.display.set_caption("Monkey Fever")
-    pg.mouse.set_visible(False)
-
-    # Create The Background
-    background = pg.Surface(screen.get_size())
-    background = background.convert()
-    background.fill((170, 238, 187))
-
-    # Put Text On The Background, Centered
-    if pg.font:
-        font = pg.font.Font(None, 64)
-        text = font.render("Pummel The Chimp, And Win $$$", True, (10, 10, 10))
-        textpos = text.get_rect(centerx=background.get_width() / 2, y=10)
-        background.blit(text, textpos)
-
-    # Display The Background
-    screen.blit(background, (0, 0))
-    pg.display.flip()
-
-    # Prepare Game Objects
-    whiff_sound = load_sound("whiff.wav")
-    punch_sound = load_sound("punch.wav")
-    chimp = Chimp()
-    fist = Fist()
-    allsprites = pg.sprite.RenderPlain((chimp, fist))
-    clock = pg.time.Clock()
-
-    # Main Loop
-    going = True
-    while going:
-        clock.tick(60)
-
-        # Handle Input Events
-        for event in pg.event.get():
-            if event.type == pg.QUIT:
-                going = False
-            elif event.type == pg.KEYDOWN and event.key == pg.K_ESCAPE:
-                going = False
-            elif event.type == pg.MOUSEBUTTONDOWN:
-                if fist.punch(chimp):
-                    punch_sound.play()  # punch
-                    chimp.punched()
-                else:
-                    whiff_sound.play()  # miss
-            elif event.type == pg.MOUSEBUTTONUP:
-                fist.unpunch()
-
-        allsprites.update()
-
-        # Draw Everything
-        screen.blit(background, (0, 0))
-        allsprites.draw(screen)
-        pg.display.flip()
-
-    pg.quit()
-
-
-# Game Over
-
-
-# this calls the 'main' function when this script is executed
+    def get_punched(self, obj):
+        """Cause monkey to spin"""
+        self.move = self._spin
+
+        
+class Game:
+
+    def initialize():
```

Comment:
The methods should have self as their first argument.

Reply:
It's effectively a static class.  So it's a static method, as well.

I couldn't justify making `Game` instantiable on this iteration but if you think it's worth the added complexity we can refactor it.

Reply:
_I'd be willing to refactor the tutorial, as well, as part of this PR._